### PR TITLE
Initial template + json schema

### PR DESCRIPTION
**Pull request by thomasyu888**
I deviated away from the original Folder structure to just contain the information in a nested `json`.  It could get hairy with a ton of folders, but I think its easier to configure.

### CODE REVIEW

**Comment by thomasyu888** [comment]
Create definitions in json schema so they can be referenced.

File: base_template.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                    {
+                        "Team": "teamB",
+                        "acl": "view"
+                    }
+                ]
+            }
+        ],
+        "Wiki": {
+            "_comment": "Use one of these below keys",
+            "WikiTemplate": "syn18058986",
```
Comment (thomasyu888) [inline base_template.json]: This should just use the `oneof` in the json schema which allows either a `syn` string or the `wikistructure`

**Review by kelshmo — COMMENTED** [review]
@thomasyu888 I'm a newbie to thinking this way so I need it spelled out a little bit :) 

- What is the relationship between `base_template.json` and `sfn_schema.json`?
- Can you provide example code of how one could implement this in a test project? For example, if someone defines "Table" in `base_template.json`, how can that json blob be passed to `synapseclient.Table`? 
- Is there specific syntax for optional parameters?

**Comment by thomasyu888** [comment]
Thanks @kelshmo for the comment.  After further discussion with @kdaily and @milen-sage, this initial template and json schema will change.  That being said, let me try to clear some things up.

* i will first start with the `sfn_schema.json`.  This is the `schema` file that describes all the components that a `json` configuration can have.  So `base_template.json` is a template that someone would configure and our initial codebase would validate this template against the json schema and also create everything specified in the `base_template.json`

* 
```        
    "Table": [
            {
                "name": "tableA",
                "description": "user description of schema",
                "annotations": {
                    "key": "pair"
                },
                "Column": [
                    {
                        "name": "column1",
                        "defaultValue": "temp",
                        "maximumSize": 1000,
                        "columnType": "STRING",
                        "facetType": "enumeration"
                    },
                    {
                        "name": "column2",
                        "defaultValue": 2,
                        "columnType": "INTEGER",
                        "facetType": "range"
                    }
                ]
            }
        ]
```
If you look at this section `base_template.json`, it actually defines a `Table`.  This particular Table has annotations `key=pair` and has a `STRING` and `INTEGER` column.  This configuration is based off of Synapse Tables are created through the rest service and client.

* The optional parameters are described in the `sfn_schema.json`.  I think I should have named `base_template.json` like `example_template.json`.  The `schema` file basically defines rules for your template.

**Pull request merged**